src/asn1_utilities.c

```diff
@@ -80,9 +81,9 @@ int xtt_asn1_from_ed25519_private_key(const xtt_ed25519_priv_key *priv_key_in,
 
     build_asn1_key_skeleton(asn1_out, &privkey_location);
 
-    int ret = xtt_crypto_extract_ed25519_private_key(privkey_location, priv_key_in);
-    if (0 != ret) {
-        return -1;
+    xtt_ecdsap256_priv_key ret = *priv_key_in;
```

Comment:
In this step, we're copying in the provided private key `priv_key_in` to the location indicated by `privkey_location`. So this should be something like
```
memcpy(privkey_location, priv_key_in, sizeof(xtt_ed25519_priv_key);
return 0;
```